src/test/java/nc/students/ayaz/controllers/VideosControllerTest.java

```diff
@@ -3,57 +3,59 @@
 import nc.students.ayaz.model.Video;
 import nc.students.ayaz.repositories.UserRepository;
-import org.json.JSONObject;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.boot.test.web.client.TestRestTemplate;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.test.context.junit4.SpringRunner;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
 
 @RunWith(SpringRunner.class)
 @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class VideosControllerTest {
 
-    @Autowired
+    @MockBean
     private UserRepository repository;
 
     @Autowired
     private TestRestTemplate restTemplate;
 
     @Test
-    public void shouldAddVideoToUserWhenPOSTmethodSent() {
-        repository.addUser(new User("Ayaz"));
+    public void shouldAddVideoToUserWhenPostMethodSent() throws Exception {
+        User user = new User("Ayaz");
+        when(repository.getUserByNickname("Ayaz")).thenReturn(user);
 
         ResponseEntity<String> response = restTemplate.postForEntity("/users/Ayaz/videos/FunnyCats", "", String.class);
 
-        User user = repository.getUserByNickname("Ayaz");
-        assertTrue(user.ownsVideo(new Video("FunnyCats")));
         assertEquals(HttpStatus.CREATED, response.getStatusCode());
+        assertTrue(user.ownsVideo(new Video("FunnyCats")));
     }
 
     @Test
-    public void shouldReturnVideoWhenSendGETrequestWithVideoName() throws Exception {
+    public void shouldReturnVideoWhenGetRequestSentWithVideoName() throws Exception {
         User user = new User("Ayaz");
-        repository.addUser(user);
-        user.addVideo(new Video("FunnyCats"));
+        Video video = new Video("FunnyCats");
+        user.addVideo(video);
+        when(repository.getUserByNickname("Ayaz")).thenReturn(user);
 
-        ResponseEntity<String> response = restTemplate.getForEntity("/users/Ayaz/videos/FunnyCats", String.class);
+        ResponseEntity<Video> response = restTemplate.getForEntity("/users/Ayaz/videos/FunnyCats", Video.class);
 
-        JSONObject jsonVideo = new JSONObject();
-        jsonVideo.put("name", "FunnyCats");
-        assertEquals(jsonVideo.toString(), response.getBody());
         assertEquals(HttpStatus.OK, response.getStatusCode());
+        Video fetchVideo = response.getBody();
+        assertEquals(video, fetchVideo);
     }
 
     @Test
-    public void shouldReturnNotFoundStatusCodeWhenTryingToGetNonexistentVideo() {
-        repository.addUser(new User("Ayaz"));
+    public void shouldReturnNotFoundStatusCodeWhenTryingToGetNonexistentVideo() throws Exception {
+        User user = new User("Ayaz");
```

Comment:
Have you considered using mocks for classes like "User" and "Video" in the controller tests ?
These classes will probably grow a lot during next few iterations. And it probably won't be as easy to construct them as it is now. Moreover, you would have to change every test where you instantiate them if you change your constructor.
Its not an issue right now, but something to consider for future :).